x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/FrameGroupIDTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class FrameGroupIDTests extends ESTestCase {
+    public void testEmptySourceFilename() {
+        String given = FrameGroupID.getBasenameAndParent("");
+        assertEquals("", given);
+    }
+
+    public void testNonPathSourceFilename() {
+        String given = FrameGroupID.getBasenameAndParent("void jdk.internal.misc.Unsafe.park(boolean, long)");
+        assertEquals("void jdk.internal.misc.Unsafe.park(boolean, long)", given);
+    }
+
+    public void testRootSourceFilename() {
+        String given = FrameGroupID.getBasenameAndParent("/");
+        assertEquals("/", given);
+    }
+
+    public void testRelativePathSourceFilename() {
+        String given = FrameGroupID.getBasenameAndParent("../src/main.c");
+        assertEquals("src/main.c", given);
+    }
+
+    public void testAbsolutePathSourceFilename() {
+        String given = FrameGroupID.getBasenameAndParent("/usr/local/go/src/runtime/lock_futex.go");
+        assertEquals("runtime/lock_futex.go", given);
+    }
+
+    public void testEmptyFunctionName() {
+        String frameGroupID = FrameGroupID.create("FEDCBA9876543210", 177863, "", "", "");
+        assertEquals("-226939920", frameGroupID);
```

Comment:
I'm on the fence here regarding the expected value as we implicitly rely that the implementation of `Objects.hashCode()` does not change. How about we expect `Objects.hash("FEDCBA9876543210", 177863)` here instead?

Reply:
I agree, it isn't ideal to have an opaque expected value or to partially recreate the implementation. I used your suggestion, so at least we are testing if the checks in `FrameGroupID.create` work as expected, which is useful to know.